### DISCUSSION TITLE
IP address of the Rust server

### DISCUSSION THREAD

**Discussion by OdmenFil** [rustdesk/rustdesk#5459] [Q&A]
I'm interested in the question why in the ps script (in the screenshot on the right) I changed the IP address and on the left, when installing, the external IP of the router-modem is automatically registered? how to make it so that the local IP address of the Rust server is automatically entered. what I want is to save the ps script with manual settings.
screenshot link
![photo_2023-08-18_14-50-53](https://github.com/rustdesk/rustdesk/assets/142785426/6f610821-4ced-46a5-a932-f07765fa204b)

**Comment by dinger1986** [comment]
This is techaholds script, is it an old one? It’s been butchered so much, I prefer this one https://rustdesk.com/docs/en/self-host/pro/faq/#can-i-deploy-rustdesk-using-powershell would need a little adapting for setting the password etc

**Reply by OdmenFil** [reply]
did not work. changed the version of the rust desk, substituted the desired values instead of the standard ones.
I tried to install Rust Desk into the system before, removed the Revo Uninstaller with registry cleaning
![rust2](https://github.com/rustdesk/rustdesk/assets/142785426/7047ea9b-4fe2-4823-958c-492c14db203d)

**Reply by dinger1986** [reply]
If you look in the script it shows you the locations for the config files, settings aren’t in the registry.

so c:\windows\servicesprofiles\localservice\appdata\roaming\rustdesk

and 

c:\users\%username%\appdata\roaming\rustdesk

**Reply by OdmenFil** [reply]
after deleting the entire rustdesk folder. fields were not filled at all
![Screenshot_11](https://github.com/rustdesk/rustdesk/assets/142785426/95de2940-911d-478a-877c-4380105f0566)

**Reply by dinger1986** [reply]
Ok, probably similar to https://github.com/techahold/rustdeskinstall/issues/47#. I’m going to rework that script

**Reply by OdmenFil** [reply]
what does he look like? I do NOT freeze. but simply does not write the parameters to the file/program
![image](https://github.com/rustdesk/rustdesk/assets/142785426/96a5ce0a-dca7-4602-8f4e-407f01816054)

0.17 ip my PC

**Reply by dinger1986** [reply]
How are you running it? Does however you are running it have permissions to write to the service profiles folder? IE as system?

**Reply by OdmenFil** [reply]
As I started to try your script, I deleted temporary folders and files (2 months ago). maybe it helped, but your script still does not work. after that I tried the old script that is generated by the program itself. I don’t understand what the problem is, but I will try this script on other PCs, I will write if the test goes well
![image](https://github.com/rustdesk/rustdesk/assets/142785426/620e8218-a85e-4eda-a234-b9af47a7405f)

**Reply by OdmenFil** [reply]
test passed successfully:
1) a RustDesk is installed on the computer
2) entered the correct data
and all this with a script
[ps script.txt](https://github.com/rustdesk/rustdesk/files/12395643/ps.script.txt)
.txt replace with .ps1
TNX

**Reply by dinger1986** [reply]
Thanks for that. I’ll compare against what’s already there and adjust the techahold script.

can you put in 3 backticks (usually top left of keyboard under esc) at the start and bottom so it shows properly?

**Reply by OdmenFil** [reply]
I realized that the script is not displayed correctly, I wrapped it in a notepad, I just need to change the extension from .txt to .ps1

**Reply by OdmenFil** [reply]
@dinger1986 I just tried it on another computer, the script installs the program on the first run, but does not enter the data, and on the second it enters the IP address and key data. i have no idea.

![image](https://github.com/rustdesk/rustdesk/assets/142785426/0994f12c-80a1-43f8-9e13-e8ec53410896)



on a clean, first install. The script runs until temporary files are deleted. and then hangs. when reinstalling it doesn't work...

**Reply by OdmenFil** [reply]
![image](https://github.com/rustdesk/rustdesk/assets/142785426/24515705-4912-4fa3-ab10-2f007e870a1d)
Start-Process "rustdesk.exe" -argumentlist "--silent-install" -wait
